### ISSUE TITLE
Floors not being randomized in MFD

### ISSUE THREAD

**Issue by ParadiddlePride** [DRE2N/DungeonsXL#657]
**Description**
I've been trying to create a dungeon that selects x amount of floors from a list, but it continuously goes down the list in it's entirety. Additionally, it replays floors even though I specified not to.
My server is 1.13.2, using the latest DXL update.

**Reproduce**
1) Make a list of multiple floors
2) Set floor count to amount smaller than # of floors in list
3) Play dungeon

**Expected behavior**
As there is only 3 floors in the list, I would expect it to skip one of the floors. Instead it plays all 3.

[//]: # (A clear and concise description of what you expected to happen.)

**Screenshots / GIFs / videos**

[//]: # (If applicable, add screenshots to help explain your problem.)

**Relevant configuration files**

[//]: # (Please paste the relevant configuration files in the line between the quotation marks:)

My dungeon config is set to the following:
```
configVersion: 1

startFloor: HauntedMansion_floor1
endFloor: HauntedMansion_floor5
floors:
  - HauntedMansion_floor2
  - HauntedMansion_floor3
  - HauntedMansion_floor4
floorCount: 2
removeWhenPlayed: true
```

**Comment by Sataniel98** [comment]
I will review the problem and, if necessary, fix it in v0.18.

**Comment by Sataniel98** [comment]
My test brought me to 1-3-2-2-5. Make sure nothing is specified on the end sign.
